src/azure-cli-core/azure/cli/core/commands/template_create.py

```diff
@@ -40,22 +43,27 @@ def wrapped(namespace):
     else:
         validator = fold_validator
 
-    quotes = '""' if platform.system() == 'Windows' else "''"
-    quote_text = '  Use {} for none.'.format(quotes) if none_flag_value else ''
-    flag_texts = {
-        new_flag_value: '  Creates new by default.{}'.format(quote_text),
-        existing_id_flag_value: '  Uses existing resource.{}'
-                                .format(quote_text),
-        none_flag_value: '  None by default.'
-    }
-    help_text = 'Name or ID of the resource.' + flag_texts[default_value_flag]
+    if not custom_help:
+        quotes = '""' if platform.system() == 'Windows' else "''"
+        quote_text = ' Use {} for none.'.format(quotes) if none_flag_value else ''
+        parent_text = ' If name specified, must also specify {}.'.format(
+            parent_option_flag or parent_name) if parent_name else ''
+        flag_texts = {
+            new_flag_value: '  Creates new by default.{}'.format(quote_text),
+            existing_id_flag_value: '  Uses existing resource.{}{}'
+                                    .format(quote_text, parent_text),
+            none_flag_value: '  None by default.'
+        }
+        help_text = 'Name or ID of the resource.' + flag_texts[default_value_flag]
+    else:
+        help_text = custom_help
 
     register_cli_argument(scope, base_name, validator=validator, help=help_text, **kwargs)
     register_cli_argument(scope, type_field_name, help=argparse.SUPPRESS, default=None)
 
 def _name_id_fold(base_name, resource_type, type_field, #pylint: disable=too-many-arguments
                   existing_id_flag_value, new_flag_value, none_flag_value,
-                  parent_name=None, parent_type=None):
+                  parent_name=None, parent_option_flag=None, parent_type=None, base_required=True):
```

Comment:
Which argument? Parent option flag lets you specify the option name so the error message will display it correctly. Otherwise you'd get 'virutal_network_name' is required instead of '--vnet-name' is required.

If you are referring to `base_required` that refers to the base property (not the parent). Since the default is true, it would only need to be specified if the base property (for example subnet) were optional.